axum/src/extract/path/mod.rs

```diff
@@ -368,14 +386,13 @@ impl IntoResponse for FailedToDeserializePathParams {
         let (status, body) = match self.0.kind {
             ErrorKind::Message(_)
             | ErrorKind::InvalidUtf8InPathParam { .. }
-            | ErrorKind::WrongNumberOfParameters { .. }
             | ErrorKind::ParseError { .. }
             | ErrorKind::ParseErrorAtIndex { .. }
             | ErrorKind::ParseErrorAtKey { .. } => (
                 StatusCode::BAD_REQUEST,
                 format!("Invalid URL: {}", self.0.kind),
             ),
-            ErrorKind::UnsupportedType { .. } => {
+            ErrorKind::WrongNumberOfParameters { .. } | ErrorKind::UnsupportedType { .. } => {
```

Comment:
`WrongNumberOfParameters` is a programmer error so it shouldn't be a `4xx` imo. I'd categorize this as a bug fix and not a breaking change but I guess thats up for debate.

Reply:
I agree with the categorization as a fix.